### PR TITLE
IDP-1416 Add support for workleap certificate store

### PR DESCRIPTION
**Pull request by starcraft66**
## Description of changes
Added support to mount Workleap's certificate store over the container's internal certificate store. The store comes from a well-known Key Vault who's information must be filled in via helm values. 

## Breaking changes
Removed the `volumes` section that was briefly introduced (without being documented) into the chart because it is redundant and less flexible than `extraVolumes`. No one is using it AFAIK.

## Additional checks
Tested using server-side dry-run on a dev cluster:

```
PS C:\Users\tristan.g-hane\src\gsoft-helm-charts> helm template --set certificateStore.enabled=true test ./charts/aspnetcore --debug | kubectl apply --dry-run=server -f -
install.go:218: [debug] Original chart version: ""
install.go:235: [debug] CHART PATH: C:\Users\tristan.g-hane\src\gsoft-helm-charts\charts\aspnetcore

poddisruptionbudget.policy/test-pdb created (server dry run)
service/test-service created (server dry run)
deployment.apps/test-deployment created (server dry run)
ingress.networking.k8s.io/test-ingress created (server dry run)
secretproviderclass.secrets-store.csi.x-k8s.io/internal-dns-keyvault created (server dry run)
```

- [x] Updated the documentation of the project to reflect the changes
- [ ] Added new tests that cover the code changes

### CODE REVIEW

File: charts/aspnetcore/Chart.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v2
 name: aspnetcore
 description: A generic Helm chart for ASP.NET Core services
-version: 1.0.4
+version: 1.0.5
```
Comment (starcraft66) [inline charts/aspnetcore/Chart.yaml:4]: There is technically a breaking change but the removed field was undocumented and I couldn't find anyone making use of it.

File: charts/aspnetcore/templates/deployment.yaml [hunk 2]
```diff
@@ -35,9 +35,6 @@ spec:
       {{- end }}
     spec:
       serviceAccountName: {{ include "aspnetcore.serviceAccountName" . }}
-      {{- if .Values.volumes }}
```
Comment (starcraft66) [inline charts/aspnetcore/templates/deployment.yaml:38]: This overrided the entire `volumes` key making it super un-flexible and even clashed with `extraVolumes` if both were set. It should never have been included.

File: charts/aspnetcore/templates/secretproviderclass.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    usePodIdentity: "false"       
+    clientID: {{ .Values.azureWorkloadIdentity.clientId }}
+    keyvaultName: {{ .Values.certificateStore.keyvaultName }}
+    objects:  |
+      array:
+        - |
+          objectName: INTERNAL_DNS_CERTIFICATE_NAME
+          objectType: cert
+    tenantId: {{ .Values.certificateStore.tenantId }}
+{{- end }}
```
Comment (guillaume-pb-workleap) [inline charts/aspnetcore/templates/secretproviderclass.yaml:18]:
```suggestion
{{- end }}

```

File: charts/aspnetcore/values.yaml [hunk 4]
```diff
@@ -180,3 +180,12 @@ extraVolumes: []
...
 extraVolumeMounts: []
+
+## Replace the container's certificate store with Workleap's certificate store which is stored in an Azure Key Vault
+## @param certificateStore.enabled Whether or not to replace the container's certificate store with Workleap's
+## @param certificateStore.tenantId The id of the azure tenant the Key Vault containing the certificate store is located in
+## @param certificateStore.keyvaultName The name of the Key Vault containing the certificate store
+certificateStore:
+  enabled: false
+  tenantId: ""
+  keyvaultName: ""
```
Comment (guillaume-pb-workleap) [inline charts/aspnetcore/values.yaml:191]:
```suggestion
  keyvaultName: ""
  
```

**Review by guillaume-pb-workleap — APPROVED** [review]

**Review by guillaume-pb-workleap — APPROVED** [review]

**Pull request merged**